### PR TITLE
UART cleaning

### PR DESCRIPTION
**Pull request by jforissier**
- Rename core/include/drivers/{uart.h -> pl011.h} and
  core/drivers/{uart.c -> pl011.c}. Use pl011_ prefix.
- Remove WITH_UART_DRV, which was used to enable pl011 on vexpress platforms
  and sunxi_uart on sunxi platform. Replace it with CFG_PL011 and
  CFG_SUNXI_UART.
- Move platform-specific (STM) code in core/arch/arm32/tee/init.c (which
  was guarded by #ifndef WITH_UART_DRV) to core/arch/arm32/plat-stm/main.c.

Signed-off-by: Jerome Forissier <jerome.forissier@linaro.org>
Tested-by: Jerome Forissier <jerome.forissier@linaro.org> (FVP)

### CODE REVIEW

File: core/arch/arm32/plat-stm/main.c [hunk 1]
```diff
@@ -78,6 +78,8 @@ void main_init(uint32_t nsec_entry)
 	struct sm_nsec_ctx *nsec_ctx;
 	size_t pos = get_core_pos();
 
+	asc_init();
```
Comment (jenswi-linaro) [inline core/arch/arm32/plat-stm/main.c]: It could be that this is too early for the STM platform, @pascal-brand-st what do you think?

Comment (pascal-brand-st-dev) [inline core/arch/arm32/plat-stm/main.c]: Yes, it could be the case. I'll check that next week.

**Comment by pascal-brand-st-dev** [comment]
Reviewed-by: Pascal Brand <pascal.brand@linaro.org>

**Comment by jenswi-linaro** [comment]
```
Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>
```

**Comment by pascal-brand-st-dev** [comment]
@jforissier I completely forgot to check on STM platform whether or not the initialization is performed too early or not. Just leave me some time to test it.

**Comment by pascal-brand-st-dev** [comment]
Jens was right. asc_init() is performed too early. Please find at https://github.com/pascal-brand-st-dev/optee_os/tree/asc_init a correct location.

With this

    Tested-by: Pascal Brand <pascal.brand@linaro.org> (STM platform)

**Comment by jforissier** [comment]
@pascal-brand-st-dev: thanks for the asc_init code. Commits squashed and review tags added.

**Pull request merged**